src/parallel/ScatterSplitter.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2022, Antti Hyvarinen <antti.hyvarinen@gmail.com>
+ * Copyright (c) 2022, Seyedmasoud Asadzadeh <seyedmasoud.asadzadeh@usi.ch>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+#ifndef OPENSMT_SCATTERSPLITTER_H
+#define OPENSMT_SCATTERSPLITTER_H
+
+#include "SimpSMTSolver.h"
+#include "SplitData.h"
+#include "SplitContext.h"
+#include "Splitter.h"
+
+#include <PTPLib/net/Channel.hpp>
+#include <PTPLib/common/Exception.hpp>
+#include <PTPLib/common/Printer.hpp>
+
+class ScatterSplitter :  public SimpSMTSolver, virtual public Splitter {
+public:
+    ScatterSplitter(SMTConfig & c, THandler & t, PTPLib::net::Channel & ch);
+    PTPLib::net::Channel & getChannel() const   { return channel; }
+    void set_syncedStream(PTPLib::common::synced_stream & ss) { syncedStream = &ss; }  //SMTS Client owns the SyncedStream and should directly set the SyncedStream in ParallelScatterSpitter
+    vec<opensmt::pair<int,int>> const &  get_solver_branch()  const  { return solverBranch; }
+    void set_solver_branch(std::string solver_branch);
```

Comment:
True...  But I guess SMTS does call the `MainSplitter`'s corresponding `solve`/`check` method, where this could be taken.  It sounds like decoupling this information as it is now is a potential source of soundness bugs.

Reply:
interpfile() is entry point, maybe we could pass solver_address to the interpfile(solver_address) so it could pass to the solve

Reply:
That could be a really good idea!

Reply:
OK, Sure, I'll do
I thought maybe you'd not like to change OpenSMT's api signatures.

Reply:
The other option would be for ScatterSplitter to get the current solver_address from channel by taking the lock ?